File: tutorials/python/tutorial_doping.py
from semiconductor_simulator.simulator import Simulator

print("Tutorial: Doping Module")
simulator = Simulator()
print("Step 1: Initialize grid")
simulator.initialize_geometry(100, 100)
print("Step 2: Run ion implantation")
simulator.run_ion_implantation(50.0, 1e15)
print("Step 3: Run diffusion")
simulator.run_diffusion(1000.0, 3600.0)
print("Step 4: Visualize dopant profile")
simulator.initialize_renderer(400, 400)
simulator.render(simulator.get_wafer())

File: tutorials/cpp/tutorial_doping.cpp
#include "../../src/cpp/core/wafer.hpp"
#include "../../src/cpp/modules/geometry/geometry_manager.hpp"
#include "../../src/cpp/modules/doping/doping_manager.hpp"
#include <memory>
#include <iostream>

int main() {
  std::cout << "Tutorial: Doping Module\n";
  auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
  GeometryManager geometry;
  std::cout << "Step 1: Initialize grid\n";
  geometry.initializeGrid(wafer, 100, 100);
  DopingManager doping;
  std::cout << "Step 2: Run ion implantation\n";
  doping.simulateIonImplantation(wafer, 50.0, 1e15);
  std::cout << "Step 3: Run diffusion\n";
  doping.simulateDiffusion(wafer, 1000.0, 3600.0);
  std::cout << "Max dopant concentration: " << wafer->getDopantProfile().maxCoeff() << " cm^-3\n";
  return 0;
}